Libraries/LibGemini/GeminiJob.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2020, The SerenityOS developers.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <LibCore/NetworkJob.h>
+#include <LibGemini/GeminiRequest.h>
+#include <LibGemini/GeminiResponse.h>
+#include <LibGemini/Job.h>
+#include <LibTLS/TLSv12.h>
+
+namespace Gemini {
+
+class GeminiJob final : public Job {
+    C_OBJECT(GeminiJob)
+public:
+    explicit GeminiJob(const GeminiRequest& request)
+        : Job(request)
+    {
+    }
+
+    virtual ~GeminiJob() override
+    {
+    }
+
+    virtual void start() override;
+    virtual void shutdown() override;
+
+protected:
```

Comment:
`protected` in a `final` class?

Reply:
I actually have no idea how `protected` and `private` differ in this context - I shamelessly stole the structure for this from `HttpsJob` in LibHTTP. If we change this here, should we address the same issue in LibHTTP?